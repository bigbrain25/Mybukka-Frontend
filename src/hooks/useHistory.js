import { useHistory as History } from 'react-router-dom';
import {
  unstable_scheduleCallback as defer,
  unstable_ImmediatePriority as ImmediatePriority
} from 'scheduler';
import { useLoadingContext } from '../context/LoadingContext';

const useHistory = () => {
  const { push: next, ...props } = History();
  const { loading } = useLoadingContext();

  const push = (url, ...data) => {
    loading(true);
    return defer(ImmediatePriority, () => {
      loading(false);
      next(url, ...data);
    });
  };

  return { push, ...props };
};

export default useHistory;
