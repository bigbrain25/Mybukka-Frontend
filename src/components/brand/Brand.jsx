import React from 'react';
import { Link } from 'react-router-dom';
import useHistory from '../../hooks/useHistory';
import Logo from './Logo';
import './brand.scss';

const Brand = () => {
  const { push } = useHistory();

  const onClick = (e) => {
    e.preventDefault();
    push('/');
  };

  return (
    <div>
      <Link onClick={onClick} className="bukka-brand" to="/"><Logo /></Link>
    </div>
  );
};

export default Brand;
