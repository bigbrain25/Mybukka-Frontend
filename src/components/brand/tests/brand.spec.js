import React from 'react';
import Brand from '../Brand';
import useHistory from '../../../hooks/useHistory';

describe('Brand component', () => {
  const wrapper = shallow(<Brand useHistory={useHistory} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
