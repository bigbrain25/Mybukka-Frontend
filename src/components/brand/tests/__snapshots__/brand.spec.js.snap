// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Brand />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="bukka-brand"
        replace={false}
        to="/"
      >
        Mybukka
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Mybukka",
        "className": "bukka-brand",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": "Mybukka",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="bukka-brand"
          replace={false}
          to="/"
        >
          Mybukka
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Mybukka",
          "className": "bukka-brand",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Mybukka",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
